mobx/lib/src/api/observable_collections/observable_map.dart

```diff
@@ -73,7 +73,8 @@ class ObservableMap<K, V>
     _context.enforceReadPolicy(_atom);
 
     _atom.reportObserved();
-    return _map[key as K?];
+    final castedKey = key as K?;
```

Comment:
Please update from master to remove these lines. We are going with a simpler approach

Reply:
I've committed a fix for this